e2e/security/security.go

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+// +build seccomp
+
+package security
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath     string `split_words:"true"`
+	TestDir     string `split_words:"true"`
+	RunDisabled bool   `default:"false"`
+}
+
+var testenv testingEnv
+var imagePath string
+
+// testSecurityUnpriv tests security flag fuctionality for singularity exec without elevated privileges
+func testSecurityUnpriv(t *testing.T) {
+	tests := []struct {
+		name          string
+		image         string
+		action        string
+		argv          []string
+		opts          e2e.ExecOpts
+		exit          int
+		expectSuccess bool
+	}{
+		// taget UID/GID
+		{
+			name:          "Set_uid",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"id", "-u", "|", "grep", "99"},
+			opts:          e2e.ExecOpts{Security: []string{"uid:99"}},
+			exit:          1,
+			expectSuccess: false,
+		},
+		{
+			name:          "Set_gid",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"id", "-g", "|", "grep", "99"},
+			opts:          e2e.ExecOpts{Security: []string{"gid:99"}},
+			exit:          1,
+			expectSuccess: false,
+		},
+		// seccomp from json file
+		{
+			name:          "SecComp_BlackList",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"mkdir", "/tmp/foo"},
+			opts:          e2e.ExecOpts{Security: []string{"seccomp:./testdata/seccomp-profile.json"}},
+			exit:          1,
+			expectSuccess: false,
+		},
+		{
+			name:          "SecComp_true",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"true"},
+			opts:          e2e.ExecOpts{Security: []string{"seccomp:./testdata/seccomp-profile.json"}},
+			exit:          0,
+			expectSuccess: true,
+		},
+		// capabilities
+		{
+			name:          "capabilities_keep_true",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"ping", "-c", "1", "8.8.8.8"},
```

Comment:
this is fine if there's no other alternative, but is there something else that can be used to test this? Going to the network and expecting it to work the same way 100% of the time is not realistic.

What are you testing here? that net_raw is not dropped? You should get the same by using the loopback address instead of Google's public DNS.